samples/app/MeshAssimp.cpp

```diff
@@ -112,32 +115,73 @@ std::string shaderFromConfig(MaterialConfig config) {
     )SHADER";
 
     if (!config.unlit) {
+        if (config.normalUV == 0) {
+            shader += R"SHADER(
+                material.normal = texture(materialParams_normalMap, getUV0()).xyz * 2.0 - 1.0;
+                material.normal.y = -material.normal.y;
+            )SHADER";
+        } else {
+            shader += R"SHADER(
+                material.normal = texture(materialParams_normalMap, getUV1()).xyz * 2.0 - 1.0;
+                material.normal.y = -material.normal.y;
+            )SHADER";
+        }
+    }
+
+    if (config.baseColorUV == 0) {
         shader += R"SHADER(
-            material.normal = texture(materialParams_normalMap, getUV0()).xyz * 2.0 - 1.0;
-            material.normal.y = -material.normal.y;
+            prepareMaterial(material);
+            material.baseColor = texture(materialParams_baseColorMap, getUV0());
+            material.baseColor.rgb *= materialParams.baseColorFactor.xyz;
+        )SHADER";
+    } else {
+        shader += R"SHADER(
+            prepareMaterial(material);
+            material.baseColor = texture(materialParams_baseColorMap, getUV1());
+            material.baseColor.rgb *= materialParams.baseColorFactor.xyz;
         )SHADER";
     }
 
-    shader += R"SHADER(
-        prepareMaterial(material);
-        material.baseColor = texture(materialParams_baseColorMap, getUV0());
-        material.baseColor.rgb *= materialParams.baseColorFactor.xyz;
-    )SHADER";
-
     if (config.alphaMode == AlphaMode::TRANSPARENT) {
         shader += R"SHADER(
             material.baseColor.rgb *= material.baseColor.a;
         )SHADER";
     }
 
     if (!config.unlit) {
-        shader += R"SHADER(
-            material.roughness = materialParams.roughnessFactor * texture(materialParams_metallicRoughnessMap, getUV0()).g;
-            material.metallic = materialParams.metallicFactor * texture(materialParams_metallicRoughnessMap, getUV0()).b;
-            material.ambientOcclusion = texture(materialParams_aoMap, getUV0()).r;
-            material.emissive = texture(materialParams_emissiveMap, getUV0());
-            material.emissive.rgb *= materialParams.emissiveFactor.rgb;
-        )SHADER";
+        if (config.metallicRoughnessUV == 0) {
+            shader += R"SHADER(
+                material.roughness = materialParams.roughnessFactor * texture(materialParams_metallicRoughnessMap, getUV0()).g;
+                material.metallic = materialParams.metallicFactor * texture(materialParams_metallicRoughnessMap, getUV0()).b;
+            )SHADER";
+        } else {
+            shader += R"SHADER(
+                material.roughness = materialParams.roughnessFactor * texture(materialParams_metallicRoughnessMap, getUV1()).g;
+                material.metallic = materialParams.metallicFactor * texture(materialParams_metallicRoughnessMap, getUV1()).b;
+            )SHADER";
+        }
+
+        if (config.aoUV == 0) {
```

Comment:
I think this would all be more readable if you did something like:

```
shader += "float2 aoUVs = " + (config.aoUV == 0 ? "getUV0()" : "getUV1()") + ";\n";
shader += "float2 emissiveUVs = " + (config.emissiveUV == 0 ? "getUV0()" : "getUV1()") + ";\n";
```

At the beginning of the `material` function. Then you can use `aoUVs` and `emissiveUVs` directly without needing all these `if` statements.